### PR TITLE
Backend submissions 3

### PR DESCRIPTION
**Pull request by ASchrader319**
Added file upload management and finished submission handling for C and Java

### CODE REVIEW

File: server/controllers/assignmentController.js
Comment (matt0125) [inline server/controllers/assignmentController.js:1]: Made a tiny change just to make the request a const apart from file_contents for readability

File: server/controllers/testcaseController.js
Comment (matt0125) [inline server/controllers/testcaseController.js:1]: - instructor check shouldn't be necessary anymore, I think it uses an is instructor middleware before getting here
  - 12-14
- console.log
  - 21
- No real error catching for file creation, not sure if we really need it tho, it should always work if the test case creation worked but still nice to have

File: server/controllers/submissionController.js [hunk 1]
```diff
@@ -23,137 +22,152 @@ var extensionToIdMap = {
...
-            }
+      }
+  
+      const validationResponse = validateSubmissionCreation(req.body);
+      if (validationResponse !== true) {
+        return res.status(400).json({ message: "Validation failed", errors: validationResponse });
+      }
+  
+      const { assignment, languages } = await Assignment.getAssignmentLangById(assignmentId);
+      const extension = fileName.split(".")[1];
```
Comment (matt0125) [inline server/controllers/submissionController.js:51]: Are we still going to use this? Or should we just get it from the assignment for now and let judge0 return an error when it doesn't compile if its the wrong language

Comment (matt0125) [reply server/controllers/submissionController.js:51]: "this" being mapping the language from the filename

File: server/controllers/submissionController.js [hunk 2]
```diff
@@ -23,137 +22,152 @@ var extensionToIdMap = {
...
+            }
 
-        const bashID = 46;
-        Submission.createTestcaseSubmission(submission.submissionid, testcase.testcaseid,
-            await Submission.postSubmission(bashID, base64EncodedScript, testcase.input, testcase.expectedoutput, testcase.maxececutiontime, base64EncodedZip));
-    });
+            //const content = await zip.generateAsync({ type: "nodebuffer" });
+            const base64EncodedZip = content.toString('base64');
+            const bashID = 46;
+            await Submission.createTestcaseSubmission(submission.submissionid, testcase.testcaseid,
```
Comment (matt0125) [inline server/controllers/submissionController.js:164]: Could turn this into a promise like the logic 104-110 but isn't necessary

File: server/models/submissionModel.js [hunk 3]
```diff
@@ -180,28 +179,32 @@ exports.updateTestcaseSubmission = async (judge0Id, status, executionTime, memor
...
+      cpu_time_limit: maxExecutionTime ? maxExecutionTime : null, // Note: This likely should not be Base64 encoded
+      additional_files: additionalFiles ? additionalFiles : null
     };
+  
 
-    console.log(`JUDGE0_URL: ${JUDGE0_URL}/submissions/?base64_encoded=false&wait=false`);
+
+    // print body vaariables
+    
+    console.log(`JUDGE0_URL: ${url}`);
```
Comment (matt0125) [inline server/models/submissionModel.js:203]: console.log

File: server/models/submissionModel.js [hunk 4]
```diff
@@ -180,28 +179,32 @@ exports.updateTestcaseSubmission = async (judge0Id, status, executionTime, memor
...
     console.log("body: ", body);
 
     const response = await axios.post(url, body, {headers});
+    console.log(response.data);
```
Comment (matt0125) [inline server/models/submissionModel.js:207]: console.log

File: server/controllers/submissionController.js (hunk 1 above)
Comment (ASchrader319) [reply server/controllers/submissionController.js:51]: > "this" being mapping the language from the filename

It'll be used eventually but probably v2.  When they allow multiple languages per

File: server/controllers/submissionController.js (hunk 2 above)
Comment (ASchrader319) [reply server/controllers/submissionController.js:164]: > Could turn this into a promise like the logic 104-110 but isn't necessary

Had issues implementing it like that with the zip buffers.

**Pull request merged**